src/output.cpp

```diff
@@ -742,6 +792,147 @@ void TextOutput::map_hist(
   out_ << std::endl;
 }
 
+void TextOutput::map_tseries(
+    BPFtrace &bpftrace,
+    const BpfMap &map,
+    uint32_t top,
+    uint32_t div,
+    const TSeriesMap &values_by_key,
+    const std::vector<std::pair<KeyType, EpochType>> &latest_epoch_by_key) const
+{
+  map_tseries_contents(
+      bpftrace, map, top, div, values_by_key, latest_epoch_by_key);
+  out_ << std::endl;
+}
+
+template <typename T>
+std::string tseries_to_str_text(BPFtrace &bpftrace,
+                                const TSeries &values,
+                                uint32_t div,
+                                EpochType last_epoch,
+                                uint64_t interval_ns,
+                                uint64_t buckets,
+                                const SizedType &inner_type)
+{
+  constexpr int graph_width = 53;
+
+  EpochType first_epoch = util::tseries_first_epoch(values,
+                                                    last_epoch,
+                                                    buckets);
+  std::ostringstream res;
+
+  if (!first_epoch) {
+    res << "<no data>" << std::endl;
```

Comment:
Why not just always display `buckets` intervals? The logic below handles missing epochs anyway.

The only weird cases are `sum` and `count`. Do we display `0` for epochs we weren't even running in?